### ISSUE TITLE
Filter visual bug

### ISSUE THREAD

**Issue by tfadsilva** [czeckd/angular-dual-listbox#94]
Hello,

So I'm currently trying to use the angular-dual-listbox in my application, but I'm having a visual bug that I am not being able to fix. Here it is:

![opera snapshot_2018-12-31_183619_localhost](https://user-images.githubusercontent.com/14255156/50565843-fe6f3800-0d2a-11e9-9608-1129326da341.png)

As you can see, everything is working as expected except the filter, which isn't showing up in the right place.

Here is the dual-listbox in my HTML file:

```html
<dual-list [source]="allProducts" [(destination)]="catalog.products" key="id" display="name" [format]="dualListFormat"
              sort="true" filter="true" height="200px">
            </dual-list>
```

And my format definition:

```javascript
dualListFormat = { add: 'Add', remove: 'Remove', all: 'All', none: 'None', direction: 'right-to-left', draggable: true, locale: undefined };
```

EDIT: added code syntax highlighting

**Comment by czeckd** [comment]
@evilmonster18 Please let me know which OS and browser are you are seeing this with.

**Comment by tfadsilva** [comment]
@czeckd 

OS: Windows 10 x64
Browser: Opera

**Comment by tfadsilva** [comment]
I just tested it out on both Chrome and Edge as well and the exact same bug happens. I don't know if it helps, but here's the HTML file corresponding to the section 'Create New Catalog' in the image I attached initially in the post.

```html
<div class="span12">
    
    <!-- CATEGORY CREATION -->
    <div class="well">
      <form class="form-horizontal" (ngSubmit)="createCatalog()" #createCatForm="ngForm">
        <h3>Create New Catalog</h3>
  
        <!-- CATALOG TITLE -->
        <div class="control-group">
          <label class="control-label" required>Title <sup>*</sup></label>
          <div class="controls">
            <input name="catalogTitle" [(ngModel)]="catalog.title" type="text" placeholder="Catalog Title" [disabled]="creatingCatalog" required>
          </div>
        </div>

        <!-- CATALOG YEAR -->
        <div class="control-group">
          <label class="control-label" required>Year <sup>*</sup></label>
          <div class="controls">
            <select class="span3" name="catalogYear" [(ngModel)]="catalog.year" [disabled]="creatingCatalog" required>
              <option *ngFor="let year of availableCatalogYears" value="{{year}}">{{year}}</option>
            </select>
          </div>
        </div>
  
        <!-- CATALOG PRODUCTS -->
        <div class="control-group" style="position: relative;">
          <label class="control-label" style="position: absolute; top: 45%;">Products </label>
          <div class="controls">
            <dual-list [source]="allProducts" [(destination)]="catalog.products" key="id" display="name" [format]="dualListFormat"
              sort="true" height="200px" [disabled]=creatingCatalog>
            </dual-list>
          </div>
        </div>
  
        <div class="control-group" style="margin-top: 40px;">
          <div class="controls">
            <input [disabled]="!createCatForm.form.valid || creatingCatalog" type="submit" name="createCatalogSubmit" value="Create Catalog"
              class="exclusive shopBtn">
            <img *ngIf="creatingCatalog" style="margin-left: 10px;"
              src="data:image/gif;base64,R0lGODlhEAAQAPIAAP///wAAAMLCwkJCQgAAAGJiYo
              KCgpKSkiH/C05FVFNDQVBFMi4wAwEAAAAh/hpDcmVhdGVkIHdpdGggYWpheGxvYWQuaW5
              mbwAh+QQJCgAAACwAAAAAEAAQAAADMwi63P4wyklrE2MIOggZnAdOmGYJRbExwroUmcG2
              LmDEwnHQLVsYOd2mBzkYDAdKa+dIAAAh+QQJCgAAACwAAAAAEAAQAAADNAi63P5OjCEgG
              4QMu7DmikRxQlFUYDEZIGBMRVsaqHwctXXf7WEYB4Ag1xjihkMZsiUkKhIAIfkECQoAAA
              AsAAAAABAAEAAAAzYIujIjK8pByJDMlFYvBoVjHA70GU7xSUJhmKtwHPAKzLO9HMaoKwJ
              Z7Rf8AYPDDzKpZBqfvwQAIfkECQoAAAAsAAAAABAAEAAAAzMIumIlK8oyhpHsnFZfhYum
              CYUhDAQxRIdhHBGqRoKw0R8DYlJd8z0fMDgsGo/IpHI5TAAAIfkECQoAAAAsAAAAABAAE
              AAAAzIIunInK0rnZBTwGPNMgQwmdsNgXGJUlIWEuR5oWUIpz8pAEAMe6TwfwyYsGo/IpF
              KSAAAh+QQJCgAAACwAAAAAEAAQAAADMwi6IMKQORfjdOe82p4wGccc4CEuQradylesojE
              MBgsUc2G7sDX3lQGBMLAJibufbSlKAAAh+QQJCgAAACwAAAAAEAAQAAADMgi63P7wCRHZ
              nFVdmgHu2nFwlWCI3WGc3TSWhUFGxTAUkGCbtgENBMJAEJsxgMLWzpEAACH5BAkKAAAAL
              AAAAAAQABAAAAMyCLrc/jDKSatlQtScKdceCAjDII7HcQ4EMTCpyrCuUBjCYRgHVtqlAi
              B1YhiCnlsRkAAAOwAAAAAAAAAAAA==" />
          </div>
        </div>
  
        <ngb-alert *ngIf="createCatalogAlert.message" [type]="createCatalogAlert.type" 
          (close)="createCatalogAlert.message = null">{{createCatalogAlert.message}}</ngb-alert>
  
      </form>
    </div>
  </div>
```

**Comment by czeckd** [comment]
If the other browsers are doing this as well, it causes me to think that additional styles in your application or lack of bootstrap3 styles may be the cause. Can you run the demo on the website without issue?

**Comment by tfadsilva** [comment]
Yes, the demo runs perfectly on the browser.

That's what I thought as well, that's why I went through the application and removed every single custom styling line of code AND added the basic bootstrap styling, and the problem still persisted

**Comment by czeckd** [comment]
I ran your template adding `filter="true"` and commented out the `ngb-alert` with the following code:
```typescript
import { Component } from '@angular/core';

@Component({
    selector: 'app-root',
    templateUrl: './app.component.html',
    styleUrls: ['./app.component.scss']
})
export class AppComponent {
    title = 'wonky';
    catalog = {
        products: [],
        year: 0,
        title: ''
    };
    allProducts: Array<any> = [
        { id: 1, name: 'foo' },
        { id: 2, name: 'bar' },
        { id: 3, name: 'goo' },
        { id: 4, name: 'har' }
    ];
    availableCatalogYears = [ 2019, 2020, 2021 ];
    dualListFormat = {};
    creatingCatalog = false;
}
```

Add this Bootstrap3 style only in the index.html:
```html
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
 integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

```
This is how it rendered in Chrome:
![issue94](https://user-images.githubusercontent.com/17859320/50622773-d4ea1400-0ec4-11e9-9f7a-47fa65e51ee6.png)


One of the other styles you are using is causing interference.  This is the template code around the filter:

```html
<form *ngIf="filter" class="filter">
  <input class="form-control" name="filterSource" [(ngModel)]="available.picker" (ngModelChange)="onFilter(available)">
</form>
```
And this is the corresponding styles:

```css
.filter {
	margin-bottom: -2.2em;
}

.filter::after {
	content:"o";
	width:40px;
	color:transparent;
	font-size:2em;
	background-image:url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path d="M0 64l192 192v192l128-32V256L512 64H0z"/></svg>');
	background-repeat:no-repeat;
	background-position:center center;
	opacity:.2;
	top: -36px;
	left: calc(100% - 21px);
	position:relative;
}
```

A guess would be that there is a global "filter" style that is causing the problem. Without seeing all the other styles being used, it's only a guess to where.

**Comment by tfadsilva** [comment]
I found out the problem. It has nothing to do with the styles I already had in the application, but the one I was missing. My application is using a bootstrap style, but it's the version 2. As soon as I imported the version 3 that you linked, it worked. But since I can't really change the global bootstrap style of the application, since it ruins the entire UI, I have to keep the one I am already using. Thank you for your help!
